lib/aba.rb

```diff
@@ -41,6 +60,30 @@ def add_transaction(transaction)
     @transactions << transaction
   end
 
+  def transactions
+    @transactions
+  end
+
+  def transactions_valid?
+    transaction_errors = {}
+    @transactions.each_with_index do |transaction, i|
+      transaction_errors[i] = transaction.errors unless transaction.valid?
+    end
+    @transaction_errors = transaction_errors if transaction_errors.length
+    @transaction_errors.empty?
+  end
+
+  # Not updated until transactions_valid? is called - simulates Validations.errors
+  def transaction_errors
+    @transaction_errors
+  end
+
+  # Updates @transaction_errors - simulates Validations.get_errors
+  def get_transaction_errors
```

Comment:
This method name and ```transaction_errors``` have the same meaning. Maybe it's better to have a ```valid?``` method for the aba instance that will validate both the aba and its transactions? It looks like we are pulling a lot of transaction-related code into aba. If someone needs to get all transaction errors they can iterate over aba.transactions